components/esp_lvgl_port/esp_lvgl_port.c

```diff
@@ -784,6 +916,323 @@ static void lvgl_port_btn_up_handler(void *arg, void *arg2)
+                    } else if (keyboard->key[i] == HID_KEY_DOWN) {
+                        key = LV_KEY_DOWN;
+                    } else if (keyboard->key[i] == HID_KEY_UP) {
+                        key = LV_KEY_UP;
+                    } else if (keyboard->key[i] == HID_KEY_ENTER || keyboard->key[i] == HID_KEY_KEYPAD_ENTER) {
+                        key = LV_KEY_ENTER;
+                    } else if (keyboard->key[i] == HID_KEY_DELETE) {
+                        key = LV_KEY_DEL;
+                    } else if (keyboard->key[i] == HID_KEY_HOME) {
+                        key = LV_KEY_HOME;
+                    } else if (keyboard->key[i] == HID_KEY_END) {
+                        key = LV_KEY_END;
+                    } else {
+                        /* Get ASCII char */
+                        key = usb_hid_get_keyboard_char(keyboard->key[i], (keyboard->modifier.left_shift || keyboard->modifier.right_shift));
+                    }
+
+                    if (key == 0) {
+                        ESP_LOGI(TAG, "Not recognized key: %c (%d)", keyboard->key[i], keyboard->key[i]);
+                    }
+                    hid_ctx->kb.last_key = key;
+                    hid_ctx->kb.pressed = true;
+                }
+            }
+
+        } else if (dev.proto == HID_PROTOCOL_MOUSE) {
+            hid_mouse_input_report_boot_t *mouse = (hid_mouse_input_report_boot_t *)data;
+            if (data_length < sizeof(hid_mouse_input_report_boot_t)) {
+                break;
+            }
+            hid_ctx->mouse.left_button = mouse->buttons.button1;
+            hid_ctx->mouse.x += mouse->x_displacement;
+            hid_ctx->mouse.y += mouse->y_displacement;
+        }
+        break;
+    case HID_HOST_INTERFACE_EVENT_TRANSFER_ERROR:
+        break;
+    case HID_HOST_INTERFACE_EVENT_DISCONNECTED:
+        hid_host_device_close(hid_device_handle);
+        break;
+    default:
+        break;
+    }
+}
+
+static void lvgl_port_usb_hid_task(void *arg)
+{
+    hid_host_dev_params_t dev;
+    lvgl_port_usb_hid_ctx_t *ctx = (lvgl_port_usb_hid_ctx_t *)arg;
+    hid_host_device_handle_t hid_device_handle = NULL;
+    vgl_port_usb_hid_event_t msg;
+
+    assert(ctx);
+
+    ctx->running = true;
+
+    while (ctx->running) {
+        if (xQueueReceive(ctx->queue, &msg, pdMS_TO_TICKS(50))) {
+            hid_device_handle = msg.hid_device_handle;
+            hid_host_device_get_params(hid_device_handle, &dev);
+
+            switch (msg.event) {
+            case HID_HOST_DRIVER_EVENT_CONNECTED:
+                /* Handle mouse or keyboard */
+                if (dev.proto == HID_PROTOCOL_KEYBOARD || dev.proto == HID_PROTOCOL_MOUSE) {
+                    const hid_host_device_config_t dev_config = {
+                        .callback = lvgl_port_usb_hid_host_interface_callback,
+                        .callback_arg = ctx
+                    };
+
+                    ESP_ERROR_CHECK( hid_host_device_open(hid_device_handle, &dev_config) );
```

Comment:
I seem to remember we have introduced some helper macros for error checking in BSPs. Could we use them here, to propagate the error to the user instead of aborting?

Reply:
lvgl_port can be used without BSP, then we cannot use BSP error checking.

Reply:
In this case, let's propagate errors to the caller, instead of aborting.

Reply:
I wanted to return errors, but here it is in task. I can add error callback and call it here, when error occur.